openpype/modules/deadline/abstract_submit_deadline.py

```diff
@@ -516,15 +517,17 @@ def from_published_scene(self, replace_in_path=True):
             published.
 
         """
-
         instance = self._instance
         workfile_instance = self._get_workfile_instance(instance.context)
         if workfile_instance is None:
             return
 
         # determine published path from Anatomy.
         template_data = workfile_instance.data.get("anatomyData")
-        rep = workfile_instance.data.get("representations")[0]
+        if self.import_reference:
+            rep = workfile_instance.data.get("representations")[1]
+        else:
+            rep = workfile_instance.data.get("representations")[0]
```

Comment:
This seems oddly hardcoded - looking at the logic [here](https://github.com/ynput/OpenPype/pull/3875/files#diff-2ccaafdd3a3dd7b8cf293860eb9bf6551427f5b38cdb63fe37a8a2101a1d5674R141-R142) I'm pretty sure we are not _that_ sure the representations will be second one?

Reply:
> When the extractor publishing the workfile, it will publish two workfiles, one is the scene with reference, the other is the scene with imported references(with the name e.g. {scene_name}_imported.ma). I set up this condition to take the second scene in order to match the submitted scene with imported reference. If not doing this, the rendering in deadline would error out.

Reply:
I understand what the if statement is trying to do. I'm merely saying that purely by looking at the code it seems quite an implicit expectation that the first will be the non-imported and the second will be the imported version. Somehow I'd have expected we should somehow be able to do a more explicit check like:
```python
representations = workfile_instance.data["representations"]
rep = next(rep for rep in representations if rep["outputName"] == "imported")
```
Consider the above pseudocode. :) 

Additionally it seems odd we're doing `workfile_instance.data.get` since we're explicitly trying to index it with `[0]` and `[1]` making the assumption that the key will actually exist. Thus I'd recommend to at least change it to:
```suggestion
        if self.import_reference:
            rep = workfile_instance.data["representations"][1]
        else:
            rep = workfile_instance.data["representations"][0]
```